src/resolve/resolved-manager.c

```diff
@@ -565,6 +565,26 @@ static int manager_memory_pressure_listen(Manager *m) {
         return 0;
 }
 
+static int dnssec_init(Manager *m) {
+        int r;
+
+        assert(m);
+
+        r = dns_trust_anchor_load(&m->trust_anchor);
+        if (r < 0)
+                return r;
+
+        r = manager_init_dnssec_algorithms(m);
+        if (r < 0)
+                return log_error_errno(r, "Failed to initialize list of allowed DNSSEC algorithms: %m");
+
+        r = manager_init_dnssec_digests(m);
+        if (r < 0)
+                return log_error_errno(r, "Failed to initialize list of allowed DNSSEC digests: %m");
```

Comment:
Most of the time, all compiled-in algorithms and digests will be allowed. So I think the semantics of the sets should be inverted: set==NULL → everything is allowed set={} → nothing is allowed, set={…} → specified algorithms are allowed. Then the init here will not be necessary and we'll save some memory and speed up lookups in the default case.

Reply:
i think i agree with @keszybz.

though an empty algorithm set doesn't really make sense hence i'd just say: if dnssec is enabled but an empty algorithm set in place this is treated as "evertyhing allowed". if people really want to refuse all algorithms they should instead just disable dnssec